workspace_host/interface.js

```diff
@@ -243,14 +245,19 @@ async.series([
             update_queue[key] = {action: 'delete'};
         });
         async function autoUpdateJobManagerTimeout() {
+            const timeout_id = setTimeout(() => {
+                logger.info(`_autoUpdateJobManager() timed out, update queue:\n${update_queue}`);
+            }, config.workspaceHostFileWatchIntervalSec * 1000);
             await _autoUpdateJobManager();
+            clearTimeout(timeout_id);
```

Comment:
Another approach to make sure `_autoUpdateJobManager()` is running is to do something like:
```
let lastRunTime = Date.now();

setInterval(() => {
    const elapsedSec = (Date.now() - lastRunTime) / 1000;
    if (elapsedSec > 30) {
        logger.error(`_autoUpdateJobManager() has not run for ${elapsedSec} seconds`);
    }
}, 1000);

# in _autoUpdateJobManager:
    lastRunTime = Date.now();
```